dissect/target/plugins/apps/webserver/apache.py

```diff
@@ -77,83 +196,165 @@ def get_log_paths(self) -> list[Path]:
+                    else:
+                        set_to_update = access_log_paths
+                        pattern_to_use = RE_CONFIG_CUSTOM_LOG_DIRECTIVE
+
+                    match = pattern_to_use.match(line)
+                    if not match:
                         self.target.log.warning("Unexpected Apache log configuration: %s (%s)", line, path)
+                        continue
+
+                    directive = match.groupdict()
+                    custom_log = self.target.fs.path(directive["location"])
+                    set_to_update.update(path for path in custom_log.parent.glob(f"{custom_log.name}*"))
 
-        return log_paths
+        return sorted(access_log_paths), sorted(error_log_paths)
 
     @plugin.export(record=WebserverAccessLogRecord)
     def access(self) -> Iterator[WebserverAccessLogRecord]:
         """Return contents of Apache access log files in unified WebserverAccessLogRecord format."""
-        for path in self.log_paths:
+        for line, path in self._iterate_log_lines(self.access_log_paths):
+            try:
+                logformat = self.infer_access_log_format(line)
+                if not logformat:
+                    self.target.log.warning("Apache log format could not be inferred for log line: %s (%s)", line, path)
+                    continue
+
+                match = logformat.pattern.match(line)
+                if not match:
+                    self.target.log.warning(
+                        "Could not match Apache log format %s for log line: %s (%s)", logformat.name, line, path
+                    )
+                    continue
+
+                log = match.groupdict()
+                if response_time := log.get("response_time"):
+                    response_time = apache_response_time_to_ms(response_time)
+
+                yield WebserverAccessLogRecord(
+                    ts=datetime.strptime(log["ts"], "%d/%b/%Y:%H:%M:%S %z"),
+                    remote_user=log["remote_user"],
+                    remote_ip=log["remote_ip"],
+                    local_ip=log.get("local_ip"),
+                    method=log["method"],
+                    uri=log["uri"],
+                    protocol=log["protocol"],
+                    status_code=log["status_code"],
+                    bytes_sent=log["bytes_sent"].strip("-") or 0,
+                    pid=log.get("pid"),
+                    referer=log.get("referer"),
+                    useragent=log.get("useragent"),
+                    response_time_ms=response_time,
+                    source=path,
+                    _target=self.target,
+                )
+
+            except Exception as e:
+                self.target.log.warning("An error occured parsing Apache log file %s: %s", path, str(e))
+                self.target.log.debug("", exc_info=e)
+
+    @plugin.export(record=WebserverErrorLogRecord)
+    def error(self) -> Iterator[WebserverErrorLogRecord]:
```

Comment:
Can you add a `error` namespace function to `webserver` too? If you're feeling daring, you could also upgrade that namespace plugin to a proper [`NamespacePlugin`](https://github.com/fox-it/dissect.target/blob/main/dissect/target/plugin.py#L823).

Reply:
Took the dare, ran into a few bugs with the `NamespacePlugin`, which is why the tests fail until https://github.com/fox-it/dissect.target/pull/463 is merged into main and this branch is rebased.